drivers/mlx90393/mlx90393_saul.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2023 Michael Ristau
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_mlx90393
+ * @{
+ *
+ * @file
+ * @brief       MLX90393 adaption to the RIOT actuator/sensor interface
+ *
+ * @author      Michael Ristau <michael.ristau@fh-erfurt.de>
+ *
+ * @}
+ */
+
+#include "saul.h"
+#include "mlx90393.h"
+
+static int read(const void *dev, phydat_t *res)
+{
+    /* Maximum value that can be achieved is +- 106480 uT according to Table 17
+    and 21 from Datasheet. To fit in signed 16 bit only the 16 MSB are passed. */
+    mlx90393_data_t data;
+    if (mlx90393_read((mlx90393_t*) dev, &data) == 0) {
+        res->val[0] = (int16_t) data.x_axis >> 2;
+        res->val[1] = (int16_t) data.y_axis >> 2;
+        res->val[2] = (int16_t) data.z_axis >> 2;
+        res->unit = UNIT_T;
+        res->scale = -6;
```

Comment:
Laut documentation liefert `mlx90393_read` die Daten in uTesla. Wenn Sie die aber durch 4 Teilen, kommt etwas komisches heraus, aber nicht uTesla.
Zudem würde in Ihrem Fall erst der Type Cast (Narrowing Conversion) und dann die Division erfolgen.

Wenn Sie ein Problem mit dem Wertebereich haben, wäre eine mögliche Lösung um möglichst wenig Genauigkeit zu verlieren aus meiner Sicht:
```suggestion
        res->val[0] = (int16_t)(data.x_axis / 10);
        res->val[1] = (int16_t)(data.y_axis / 10);
        res->val[2] = (int16_t)(data.z_axis / 10);
        res->unit = UNIT_T;
        res->scale = -5;
```
Oder alternativ gleich in mTesla:
```suggestion
        res->val[0] = (int16_t) data.x_axis / 1000;
        res->val[1] = (int16_t) data.y_axis / 1000;
        res->val[2] = (int16_t) data.z_axis / 1000;
        res->unit = UNIT_T;
        res->scale = -3;
```
Dazu fehlt mir aber das Gefühl in welchem Wertebreich sich die Werte bewegen und wie genau das Ergebnis sein muss.